### PR TITLE
pioneer 20150810 (new formula)

### PR DESCRIPTION
**Pull request by tomyun**
[Pioneer](http://pioneerspacesim.net/) is a space adventure game being [actively developed](https://github.com/pioneerspacesim/pioneer).

By the way, it's not to be confused with [Pioneers](http://pio.sourceforge.net/), a Catan game, that we already have a formula.

### CODE REVIEW

File: pioneer.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  depends_on "sdl2"
+  depends_on "sdl2_image"
+  depends_on "libsigc++"
+  depends_on "libvorbis"
+  depends_on "libpng"
+  depends_on "lua"
+
+  def install
+    ENV.cxx11
+    ENV["PIONEER_DATA_DIR"] = "#{pkgshare}/data"
```
Comment (dunn) [inline pioneer.rb]: Would it be more standard to put application data in `var/"pioneer"`?

**Comment by dunn** [comment]
The build failed because of a test-bot error, but I also wrote a formula for this a few weeks ago and had to add
```ruby
    ENV["GL_CFLAGS"] = "-I#{MacOS.sdk_path}/System/Library/Frameworks/OpenGL.framework/Headers"
    ENV["GL_LIBS"] = "-I#{MacOS.sdk_path}/System/Library/Frameworks/OpenGL.framework/Libraries"
```
in the install block, or the game wouldn't launch.

File: pioneer.rb (hunk 1 above)
Comment (tomyun) [inline pioneer.rb]: I thought `pkgshare` would better fit if the data remains unchanged after install, meanwhile `var` was more for something that can be changed or handled by users. Looks like `PIONEER_DATA_DIR` is static and `~/Library/Application Support/Pioneer` is created for saving user data.

Comment (dunn) [inline pioneer.rb]: Oh, got it.  That makes sense, then.

**Comment by tomyun** [comment]
On my machine running on Yosemite, it built and ran fine without explicit `GL` flags:

```
~ otool -L /usr/local/Cellar/pioneer/20150810/bin/pioneer
/usr/local/Cellar/pioneer/20150810/bin/pioneer:
	...
	/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib (compatibility version 1.0.0, current version 1.0.0)
	...
```

I found `configure` complains missing `gl` package for `pkg-config`, but it seems to work fine with default settings at least for me. Did you see any build failure without them?

And `ENV.cxx11` had to be put in due to `cinttypes` error found with a build bot running on Mountain Lion. After pushing this fix, CI bots somewhat went wrong and failed with a tons of Jenkins error. It'd be nice if one can clean up and restart the bots. The same issue has happened quite often recently.

**Comment by dunn** [comment]
Oh yeah, if `ENV.cxx11` is for ML that makes sense.

Restarted the build.

**Comment by mistydemeo** [comment]
@BrewTestBot test this please

**Comment by mistydemeo** [comment]
@BrewTestBot test this please

**Comment by tomyun** [comment]
Can we clean up CI bots and rerun the test? It seems the bots get stuck if forced commits are pushed in while they are working on the same branch.

Actually, a new release (20150903) just came out today. I'll make another PR once it's merged.

**Comment by dunn** [comment]
Sometimes the bots just glitch.  If you update to the latest version and push again, it'll trigger a new build.

**Comment by tomyun** [comment]
@dunn I tried pushing it again multiple times but failed with the same issue. Let's see how it turns out this time.

**Comment by tomyun** [comment]
@dunn It didn't help this time either. Any idea how we can move on? I think there were several cases resolved by manually triggered bots, probably after some cleanup in the server-side?

**Pull request closed without merging**